app/src/main/java/com/github/se/polyfit/data/remote/firebase/MealFirebaseRepository.kt

```diff
@@ -0,0 +1,77 @@
+package com.github.se.polyfit.data.remote.firebase
+
+import android.util.Log
+import com.github.se.polyfit.model.meal.Meal
+import com.google.android.gms.tasks.Task
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.FirebaseFirestore
+
+class MealFirebaseRepository(
+    private val userId: String,
+    private val db: FirebaseFirestore = FirebaseFirestore.getInstance()
+) {
+
+  private val mealCollection = db.collection("users").document(userId).collection("meals")
+
+  fun storeMeal(meal: Meal): Task<DocumentReference> {
+    val mealData = Meal.serialize(meal)
+    return if (meal.firebasaeId.isNotEmpty()) {
+      mealCollection.document(meal.firebasaeId).set(mealData).continueWith { task ->
+        if (task.isSuccessful) {
+          mealCollection.document(meal.firebasaeId)
+        } else {
+          throw task.exception!!
+        }
+      }
+    } else {
+      mealCollection.add(mealData).continueWith { task ->
+        if (task.isSuccessful) {
+          meal.firebasaeId = task.result!!.id
```

Comment:
if its a new meal, then the firebaseId is the `task.result!!.id`, but then the meal we just added wouldn't have firebaseId field set, so then to update it would `document(meal.firebasaeId)` be found?

Reply:
So, if a meal does not have a Firebase ID, it is stored using mealCollection.add, which returns the document ID (FirebaseID). If a meal already has an ID, then it updates this location with the new values. The Firebase ID is not stored as arguments in the database because it is equal to the document ID. This avoids storing the same value twice.